src/test/java/org/jitsi/meet/test/base/AbstractParticipantHelper.java

```diff
@@ -0,0 +1,177 @@
+
+    /**
+     * Starts the owner and the seconds participant, if they are not started,
+     * and stops the third participant, if it is not stopped.
+     * participants(owner and 'second participant').
+     */
+    public void ensureTwoParticipants()
+    {
+        waitForParticipant1ToJoinMUC();
+        waitForSecondParticipantToConnect();
+
+        // TODO do we need this???
+        //closeThirdParticipant();
+    }
+
+    /**
+     * Waits until the owner joins the room, creating and starting the owner
+     * if it hasn't been started.
+     */
+    public void waitForParticipant1ToJoinMUC()
+    {
+        if (participant1 == null)
+        {
+            participant1
+                = ParticipantFactory.getInstance()
+                    .createParticipant("web.participant1");
+        }
+
+        if (participant1.isHungUp())
+        {
+            // join room
+            participant1.joinConference(currentRoomName);
+        }
+
+        MeetUtils.waitForParticipantToJoinMUC(participant1.getDriver(), 15);
+    }
+
+    /**
+     * Setup helper by generating the room name to use.
+     */
+    public void setup()
+    {
+        this.currentRoomName
+            = "torture" + String.valueOf((int)(Math.random()*1000000));
+    }
+
+    /**
+     * Cleans up by quiting all participants.
+     */
+    public void cleanup()
+    {
+        quitParticipant(participant1);
+        quitParticipant(participant2);
+        quitParticipant(participant3);
+    }
+
+    /**
+     * Quits a participant if initialized.
+     * @param participant the participant to quit.
+     */
+    private void quitParticipant(Participant<WebDriver> participant)
+    {
+        if (participant == null)
+            return;
+
+        participant.quit();
+    }
+
+    /**
+     * Waits until {@code secondParticipant} has joined the conference (its ICE
+     * connection has completed and has it has sent and received data).
+     */
+    public void waitForSecondParticipantToConnect()
+    {
+        if (participant2 == null)
+        {
+            participant2 = ParticipantFactory.getInstance()
+                .createParticipant("web.participant2");
+        }
+
+        if (participant2.isHungUp())
+        {
+            participant2.joinConference(currentRoomName);
+        }
+
+        MeetUtils.waitForParticipantToJoinMUC(participant2.getDriver(), 10);
+        MeetUtils.waitForIceConnected(participant2.getDriver());
+        MeetUtils.waitForSendReceiveData(participant2.getDriver());
+
+        TestUtils.waitMillis(5000);
+    }
+
+    /**
+     * Starts the owner, if it isn't started and hangups all other participants.
+     */
+    public void closeAllParticipantsExceptTheOwner()
+    {
+        waitForParticipant1ToJoinMUC();
+
+        if (participant2 != null && !participant2.isHungUp())
+        {
+            participant2.hangUp();
+        }
+
+        if (participant3 != null && !participant3.isHungUp())
```

Comment:
This conditions ask for a method "hangupXParticipant". It feels like it's a logic that would be needed in more places, but I haven't check that yet.